app/routes.js

```diff
@@ -372,18 +374,20 @@ router.get('/preview-1/:datasetname/:datafileid', function (req, res) {
             var str="";
             response.on('data', data => {
               str += data;
-              // If we've got more than 32000 bytes
-              if (str.length>32000) {
+              var lines = Math.floor(response.headers['content-length']/263)
+              // If we've got more than 10000 bytes
+              if (str.length>10000) {
                 str = str.split('\n').slice(0,6).join('\n');
```

Comment:
At this point (well, after this line has run), we have a string that contains the content for the 5 lines we display, and we're just about to parse those lines into a different format.  we could use str.length to find out how long those 5 lines are, so we could guess that one line is str.length/5.  Given the content-length, and the length of one line, we could guestimate how many lines there are in total.